### PR TITLE
CU-866aurm8v - Create Register Page and connect with Firebase

### PR DESCRIPTION
**Pull request by vaionexbusra**
**Task:**

[Link](https://app.clickup.com/t/866aurm8v)

**What have done:**

Login ve register page created
Register page ui added
Login page ui added
shad-cn added
toastify replaced with shad-cn toast
Context created
signUp and signIn functions added

Screenshot:

image
image
image

### CODE REVIEW

File: .env.local
Comment (paulsp94) [inline .env.local:1]: I dont think we wanna leak secrets here

File: src/app/layout.js
Comment (paulsp94) [inline src/app/layout.js:1]: lets review the impact on page load time of pulling in firebase in root, we might wanna lazy load it

File: src/components/Layout/header/header.js [hunk 1]
```diff
@@ -51,10 +52,17 @@ export function Header() {
...
         <div className="hidden lg:flex lg:flex-1 lg:justify-end">
-          <Link href="#" className="text-sm font-semibold leading-6 text-white">
-            Log in
-            {' '}
-            <span aria-hidden="true">&rarr;</span>
+          <Link
+            href="/login"
+            className="text-sm font-semibold leading-6 mx-3 text-white"
+          >
+            Log in <span aria-hidden="true">&rarr;</span>
```
Comment (paulsp94) [inline src/components/Layout/header/header.js:59]: whats that can we just delete it? same for below

File: src/components/auth/Login/index.jsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    placeholder: '',
+  },
+  {
+    label: 'Password',
+    name: 'password',
+    type: 'password',
+    placeholder: '',
+  },
+]
+export default function LoginComponent() {
```
Comment (paulsp94) [inline src/components/auth/Login/index.jsx:27]: NIT: new line would be nice it our prettier not doing that?

File: src/components/auth/Login/index.jsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    password: '',
+  })
+  const { push } = useRouter()
+  const [isChecked, setIsChecked] = useState(true)
+
+  const handleChange = (e) => {
+    const { name, value } = e.target
+    setFormData((prevData) => ({ ...prevData, [name]: value }))
+  }
+  const handleEmailSignIn = async () => {}
```
Comment (paulsp94) [inline src/components/auth/Login/index.jsx:40]: this looks incomplete

File: src/components/auth/Register/index.jsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    name: 'confirmPassword',
+    type: 'password',
+    placeholder: '',
+  },
+]
+
+export default function RegisterComponent() {
+  const { push } = useRouter()
+  const { authUser, updateUserData } = useFirebaseAuthContext()
+  const [formData, setFormData] = useState({
```
Comment (paulsp94) [inline src/components/auth/Register/index.jsx:48]: we might wanna look into using react-forms or formik

File: src/contexts/firebaseAuthContext.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+'use client'
+
+import React, { createContext, useContext } from 'react'
+import { useFirebaseAuth } from '@/firebase/auth'
+import { ToastContainer } from 'react-toastify'
```
Comment (paulsp94) [inline src/contexts/firebaseAuthContext.js:5]: didnt you replace toastify?

File: src/firebase/firestore.js [hunk 6]
```diff
@@ -1,5 +1,34 @@
...
+      console.error('Error creating user document', error)
+      return null
+    }
+  }
+
+  const userDocument = await getDoc(userRef)
+  return { uid: user.uid, ...userDocument.data() }
+}
+
+// const { user } = await createUserWithEmailAndPassword(auth, email, password);
```
Comment (paulsp94) [inline src/firebase/firestore.js:33]: lets clean up these comments

**Review by paulsp94 — CHANGES_REQUESTED** [review]

File: src/app/layout.js
Comment (vaionexbusra) [reply src/app/layout.js:1]: haven't check it yet. it is possible to create anothet ticket for it?

File: src/components/Layout/header/header.js (hunk 1 above)
Comment (vaionexbusra) [reply src/components/Layout/header/header.js]: It deleted.

File: src/components/auth/Login/index.jsx (hunk 3 above)
Comment (vaionexbusra) [reply src/components/auth/Login/index.jsx]: necessary functions added.

File: src/contexts/firebaseAuthContext.js (hunk 5 above)
Comment (vaionexbusra) [reply src/contexts/firebaseAuthContext.js]: it replaced atm.

File: src/firebase/firestore.js (hunk 6 above)
Comment (vaionexbusra) [reply src/firebase/firestore.js]: ımm haven't check createUserAndFetchDocument function working or not. but for this part authentication is working fine. I will work on firestore part in different task.

File: src/app/(dashboard)/layout.js [hunk 7]
```diff
@@ -1,10 +1,18 @@
...
 export const metadata = {
   description: 'Dashboard',
   title: 'Dashboard',
 }
 
 export default function DashboardRootLayout({ children }) {
-  return <>{children}</>
+  return (
+    <>
+      {' '}
```
Comment (paulsp94) [inline src/app/(dashboard)/layout.js:12]: delete

File: src/app/layout.js [hunk 8]
```diff
@@ -1,18 +1,16 @@
...
+  description: 'Main root description of the Project',
 }
 
 export default function AppLayout({ children }) {
   return (
     <html lang="en" className={`bg-white`}>
       <body className="flex flex-col">
-        {children}
-        <ToastContainer />
+        <FirebaseAuthProvider>{children}</FirebaseAuthProvider>
```
Comment (paulsp94) [inline src/app/layout.js:13]: NIT: can you test the page load time with that, perhaps we can optimize that if we find a way to lazy load auth

**Review by paulsp94 — APPROVED** [review]
LGTM

**Pull request merged**